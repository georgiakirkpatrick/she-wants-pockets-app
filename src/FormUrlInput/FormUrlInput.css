.FormUrlInput {
    width: 100%;
    margin-top: 10px;
}

.FormUrlInput input {
    width: 100%;
    margin: 2px 0px 10px;
    padding: 10px;
    font: inherit;
}