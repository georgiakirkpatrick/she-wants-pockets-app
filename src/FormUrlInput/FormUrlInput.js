import React from 'react'
import './FormUrlInput.css'

const FormUrlInput = props => {
    return (
        <div className='FormUrlInput'>
            <label htmlFor={props.id}>{props.prompt}</label>
            <input 
                required
                type='url'
                id={props.id} 
                name={props.name}
                defaultValue={props.currentValue} 
                onChange={props.handleChange}  
            />
        </div>
    )
}

FormUrlInput.defaultProps = {
    id: 1,
    prompt: '',
    name: '',
    currentValue: '',
    onChange: () => {}
}

export default FormUrlInput

