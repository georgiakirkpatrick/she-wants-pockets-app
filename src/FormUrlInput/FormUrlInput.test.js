import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import FormUrlInput from './FormUrlInput'

describe('FormUrlInput', () => {
    it ('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<FormUrlInput />, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it ('render the UI as expected', () => {
        const tree = renderer
            .create(<FormUrlInput />)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})