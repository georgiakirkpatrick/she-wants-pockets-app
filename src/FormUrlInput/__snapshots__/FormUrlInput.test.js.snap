// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormUrlInput render the UI as expected 1`] = `
<div
  className="FormUrlInput"
>
  <label
    htmlFor={1}
  >
    
  </label>
  <input
    defaultValue=""
    id={1}
    name=""
    required={true}
    type="url"
  />
</div>
`;
